packages/frontend/tests/unit/service/channels/utils/-vote-sorter-test.ts

```diff
@@ -0,0 +1,153 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+
+import { clearLocalStorage, getStore } from 'emberclear/tests/helpers';
+import { VOTE_ACTION } from 'emberclear/models/vote-chain';
+import {
+  generateSortedVote,
+  VOTE_ORDERING,
+  SortedVote,
+  SortedVoteHex,
+} from 'emberclear/services/channels/-utils/vote-sorter';
+import {
+  convertObjectToUint8Array,
+  convertUint8ArrayToObject,
+  fromHex,
+} from 'emberclear/utils/string-encoding';
+import { buildUser } from 'emberclear/tests/helpers/factories/user-factory';
+import { equalsUint8Array } from 'emberclear/utils/uint8array-equality';
+
+module('Unit | Service | channels/utils/vote-sorter', function (hooks) {
+  setupTest(hooks);
+  clearLocalStorage(hooks);
+
+  module('vote is sorted properly', function () {
+    test('when ran with previous vote', async function (assert) {
+      const store = getStore();
+      let yes1 = await buildUser('yes1');
+      let yes2 = await buildUser('yes2');
+      let no1 = await buildUser('no1');
+      let no2 = await buildUser('no2');
+      let remaining1 = await buildUser('remaining1');
+      let remaining2 = await buildUser('remaining2');
+      let currentUser = await buildUser('currentUser');
+      let firstVote = store.createRecord('vote-chain', {
+        signature: convertObjectToUint8Array('firstVoteSignature'),
+      });
+      let currentVote = store.createRecord('vote-chain', {
+        previousVoteChain: firstVote,
+        action: VOTE_ACTION.ADD,
+        key: currentUser,
+        target: currentUser,
+        yes: [yes1, yes2],
+        no: [no1, no2],
+        remaining: [remaining2, remaining1],
+      });
+
+      let resultHex = convertUint8ArrayToObject<SortedVoteHex>(generateSortedVote(currentVote));
+      let result: SortedVote = [
```

Comment:
Maybe a Todo for later would be to add a method that does this to the sort util?

Reply:
Sort of addressed in 2c7d500. If other things outside the test need it in the future it can be moved elsewhere, however I don't see anywhere else needing it.